### PR TITLE
Improve field validations

### PR DESCRIPTION
**Pull request by netorodrigues**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| **Issue**         | https://mundipagg.atlassian.net/browse/PAOP-269
| **What?**         | Improve validations in some fields informed by the buyer.

**Notes**
- This PR contains the same changes made in Pagar.me's one: https://github.com/pagarme/ecommerce-module-core/pull/4
- This PR also adds some logs in charge status, because of an issue analyzing another case (https://mundipagg.atlassian.net/browse/PAOP-254). Everytime a charge status change, we'll se this message:

#### :package: Attachments (if appropriate)
Add additional informations like screenshots, issue link, etc

#### :speech_balloon: Important guidelines

* Add pull request labels.
* Check base branch.
* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -0,0 +1,2297 @@
+{
```
Comment (sfwill-dev) [inline composer.lock]: Esse arquivo `composer.lock` não tem que estar no commit, ele hoje não fica no nosso repo.

File: src/Kernel/Services/OrderService.php [hunk 2]
```diff
@@ -431,7 +442,7 @@ private function wasOrderChargedSuccessfully($response)
     private function persistListChargeFailed($response)
     {
         if (empty($response['charges'])) {
-            return;
+            return [];
```
Comment (sfwill-dev) [inline src/Kernel/Services/OrderService.php]: Essa mudança não foi feita nesse método no PR https://github.com/pagarme/ecommerce-module-core/pull/4, mas sim no método `createChargesFromResponse`.

File: tests/Payment/Aggregates/AddressTests.php [hunk 3]
```diff
@@ -10,16 +10,80 @@ class AddressTests extends TestCase
...
-    private $andress;
+    private $address;
 
     public function setUp()
     {
-        $this->andress = new Address();
+        $this->address = new Address();
     }
 
     public function testAddressNumberRemoveComma()
```
Comment (sfwill-dev) [inline tests/Payment/Aggregates/AddressTests.php]: Esse teste se chama `testCityTrim` no PR pagarme/ecommerce-module-core#4.

**Review by sfwill-dev — CHANGES_REQUESTED** [review]

**Comment by sfwill-dev** [comment]
Codeclimate acusou erro, mas entendo que é ok por ser hotfix e o repo da mundipagg.

File: tests/Payment/Aggregates/AddressTests.php (hunk 3 above)
Comment (netorodrigues) [reply tests/Payment/Aggregates/AddressTests.php]: feito

File: src/Kernel/Services/OrderService.php (hunk 2 above)
Comment (netorodrigues) [reply src/Kernel/Services/OrderService.php]: feito

File: composer.lock (hunk 1 above)
Comment (netorodrigues) [reply composer.lock]: arquivo removido

**Review by sfwill-dev — APPROVED** [review]

**Review by yanpantoja — APPROVED** [review]

**Pull request merged**